ui-tests/src/test/java/org/openmrs/reference/ManageProviderSchedulesTest.java

```diff
@@ -1,59 +1,64 @@
+import org.openmrs.reference.page.AppointmentSchedulingPage;
+import org.openmrs.uitestframework.test.TestData;
 
 /**
- * Created by nata on 17.07.2015.
+ * This class is meant to create the preconditions (set-up) and to clean up the environment (tear
+ * down) after the tests AddEditAppointmbetBlockTest and DeleteAppointmentBlocktTest have been
+ * executed. Subclassing ManageProviderSchedulsTest is a possible solution for a common set-up.
+ * Other possibilities are Rules or extending TestSetup. The latter possibilities might be applied
+ * in future if the test code become hard to maintain/read
  */
-public class ManageProviderSchedulesTest extends TestBase {
-    private HomePage homePage;
-    private HeaderPage headerPage;
-    private AppointmentBlocksPage appointmentBlocksPage;
-
-    @Before
-    public void setUp() throws Exception {
-        
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        headerPage = new HeaderPage(driver);
-        appointmentBlocksPage = new AppointmentBlocksPage(page);
-    }
-
-    @Ignore //ignored due to RA-904
-    @Test
-    public void manageProviderSchedulesTest() throws Exception {
-        appointmentBlocksPage.goToAppointmentBlock();
-        appointmentBlocksPage.selectLocation("Isolation Ward");
-        appointmentBlocksPage.clickOnCurrentDay();
-        appointmentBlocksPage.selectLocationBlock("Isolation Ward");
-        appointmentBlocksPage.enterStartTime("08");
-        appointmentBlocksPage.enterService("gyne");
-        appointmentBlocksPage.enterProvider("Super User");
-        appointmentBlocksPage.clickOnSave();
-        assertNotNull("Gynecology", appointmentBlocksPage.CURRENT_DAY);
-        appointmentBlocksPage.findBlock();
-        appointmentBlocksPage.clickOnEdit();
-        appointmentBlocksPage.enterService("derma");
-        appointmentBlocksPage.clickOnSave();
-        assertNotNull("Dermatology", appointmentBlocksPage.CURRENT_DAY);
-        appointmentBlocksPage.findBlock();
-        appointmentBlocksPage.clickOnDelete();
-        appointmentBlocksPage.clickOnConfirmDelete();}
-
-    @After
-    public void tearDown() throws Exception {
-        headerPage.clickOnHomeIcon();
-        headerPage.logOut();
-    }
+public class ManageProviderSchedulesTest extends ReferenceApplicationTestBase {
+	
+	protected HomePage homePage;
+	
+	protected AppointmentBlocksPage appointmentBlocksPage;
+	
+	protected String locationName;
+	
+	protected String locationUuid;
+	
+	protected String provider;
+	
+	@Before
+	public void setUp() throws Exception {
+		homePage = new HomePage(page);
+		assertPage(homePage);
+		AppointmentSchedulingPage appointmentSchedulingPage = homePage.goToAppointmentScheduling();
+		appointmentBlocksPage = appointmentSchedulingPage.goToAppointmentBlock();
+		/*
+		 * TODO The following code should be re-nabled when bug AM-186 is fixed
+		 * Location might not be available when running this test, thus we create one
```

Comment:
Sorry, I can't understand. I can see the [file ](url
![image](https://cloud.githubusercontent.com/assets/6738090/24448865/b75d71d4-1475-11e7-8228-57662b5e90d1.png)
)